test/extended/openstack/ocpbug_1765.go

```diff
@@ -0,0 +1,153 @@
+
+		g.By("preparing the openstack client")
+		networkClient, err = client.GetServiceClient(ctx, openstack.NewNetworkV2)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Failed to build the OpenStack client")
+	})
+
+	g.Context("ocpbug_1765:", func() {
+		// Automation for https://issues.redhat.com/browse/OCPBUGS-1765
+		g.It("[Serial] noAllowedAddressPairs on one port should not affect other ports", func(ctx g.SpecContext) {
+			g.By("Fetching worker machineSets")
+			var rawBytes []byte
+			var newProviderSpec machinev1alpha1.OpenstackProviderSpec
+			var rclient runtimeclient.Client
+
+			machineSets, err := getMachineSets(ctx, dc)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error getting the workers machinesets")
+
+			if len(machineSets) == 0 {
+				e2eskipper.Skipf("Expects at least one worker machineset. Found none.")
+			}
+
+			cfg, err := e2e.LoadConfig()
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error getting cluster config")
+
+			rclient, err = runtimeclient.New(cfg, runtimeclient.Options{})
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error creating a runtime client")
+
+			// Create an additional network
+			extraNetworkName := fmt.Sprintf("%v-%v", "foonet", RandomSuffix())
+			networkCreateOpts := networks.CreateOpts{Name: extraNetworkName}
+			extraNetwork, err := networks.Create(ctx, networkClient, networkCreateOpts).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error creating network")
+			g.By(fmt.Sprintf("Network %v was created", extraNetwork.Name))
+			defer networks.Delete(ctx, networkClient, extraNetwork.ID)
+
+			g.By("Create a new machineSet with an additional network with NoAllowedAddressPairs set to true")
+			// Create a machineset and add the extra network to a new machineset
+			err = machinev1.AddToScheme(scheme.Scheme)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to add Machine to scheme")
+			err = configv1.AddToScheme(scheme.Scheme)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to add Config to scheme")
+			newMachinesetParams := framework.BuildMachineSetParams(ctx, rclient, 1)
+			rawBytes, err = json.Marshal(newMachinesetParams.ProviderSpec.Value)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error marshaling new MachineSet Provider Spec")
+			err = json.Unmarshal(rawBytes, &newProviderSpec)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Error unmarshaling new MachineSet Provider Spec")
+			newMachinesetParams.Name = fmt.Sprintf("%v-%v", "extra-network", RandomSuffix())
+			newProviderSpecJson, err := json.Marshal(newProviderSpec)
+			o.Expect(err).NotTo(o.HaveOccurred(), "Failed to marshal new Machineset provider spec")
+			newMachinesetParams.ProviderSpec.Value.Raw = newProviderSpecJson
+
+			// Set NoAlloedAddressPairs for the additional network to true
+			var extraNetworkParam machinev1alpha1.NetworkParam
+			extraNetworkParam.NoAllowedAddressPairs = true
+			extraNetworkParam.Filter.ID = extraNetwork.ID
+			newProviderSpec.Networks = append(newProviderSpec.Networks, extraNetworkParam)
```

Comment:
is this really an extra network? I mean, does it really exist another network in the machine before this one is added?

Reply:
It's based on an existing worker. So it should have what the worker has + 1